src/preprocess.rs

```diff
@@ -68,20 +68,19 @@ pub fn parse_and_preprocess(path: &Path, run_opt: &RunOpt, emitter: &mut Emitter
 /// Verifies imports and canonicalizes their paths
 ///
 /// Returns None if any of imports were not recognized
-fn resolve_imports(source_path: &Path, imports: Vec<syntax::Import>, emitter: &mut Emitter)
-    -> Result<Option<Vec<PathBuf>>, Error>
+fn resolve_imports(source_path: &Path, mut imports: Vec<syntax::Import>, emitter: &mut Emitter)
```

Comment:
How can we avoid using mut here?

Reply:
```rust
let mut imports = imports;
```
in the first line.

Anyway, why do we want to remove this `mut`? (It's actually just a syntax sugar for this ↑ (in contrary to `&mut`, which changes a lot)).

Anyway... I think I don't like merging "import as a syntax element" and "import with resolved path" into one thing. We could add `preprocess::Import` as a wrapper of `syntax::Import`, but the simplest solution would be to just keep a list of resolved imports as `Map<PathBuf, syntax::Import>` (this would also remove the necessity of checking for `None` which I've seen a few times).